internal/controller/sync.go

```diff
@@ -352,19 +357,127 @@ func (r *Route) buildNextCR(ctx context.Context, route *routev1.Route, revision
 			uriSans = append(uriSans, ur)
 		}
 	}
+	var emailAddresses []string
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.EmailsAnnotationKey) {
+		emailAddresses = strings.Split(route.Annotations[cmapi.EmailsAnnotationKey], ",")
+	}
+	var organizations []string
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.SubjectOrganizationsAnnotationKey) {
+		subjectOrganizations, err := cmutil.SplitWithEscapeCSV(route.Annotations[cmapi.SubjectOrganizationsAnnotationKey])
+		organizations = subjectOrganizations
+
+		if err != nil {
+			r.log.V(1).Error(err, "the organizations annotation is invalid",
+				"object", route.Namespace+"/"+route.Name, cmapi.SubjectOrganizationsAnnotationKey,
+				route.Annotations[cmapi.SubjectOrganizationsAnnotationKey])
+			r.eventRecorder.Event(route, corev1.EventTypeWarning, ReasonInvalidValue, "annotation "+cmapi.SubjectOrganizationsAnnotationKey+": "+route.Annotations[cmapi.SubjectOrganizationsAnnotationKey]+" value is malformed")
```

Comment:
**Suggestion:** It seems like `buildNextCR` already logs the returned error. wouldn't we duplicate that log line with this? Also, wouldn't we be better off firing an event outside of `buildNextCR`?

In `sync.go`, I'd just add `r.eventRecorder.Event` like this:

```go
log.V(5).Info("route has no matching certificate request", "revision", revision)
var cr *cmapi.CertificateRequest
cr, err = r.buildNextCR(ctx, route, revision)
if err != nil {
	log.V(1).Error(err, "error generating certificate request", "object", req.NamespacedName)
    r.eventRecorder.Event(route, corev1.EventTypeWarning, CannotCreateCR, err.Error())
	// Not a reconcile error, so don't retry this revision.
	return result, nil
}
```

I'd also adjust the `err` that gets returned so that it becomes more helpful:

```suggestion
			err = fmt.Errorf("the value of the annotation %s is malformed: %w", cmapi.SubjectOrganizationsAnnotationKey, err)
```

Reply:
I tried refactoring this, but the changes are too big to fit this PR. Let's do that later.